velox/exec/Operator.h

```diff
@@ -236,7 +236,19 @@ class Operator : public BaseRuntimeStatWriter {
     virtual std::unique_ptr<Operator> toOperator(
         DriverCtx* FOLLY_NONNULL ctx,
         int32_t id,
-        const core::PlanNodePtr& node) = 0;
+        const core::PlanNodePtr& node) {
+      return nullptr;
+    }
+
+    // An overloaded method that should be called when the operator needs an
```

Comment:
It was nice to have a pure virtual method that forces the sub-class to provide an implementation. Consider keeping it that way and add exchangeClient parameter with a comment explaining that exchangeClient is non-null only if planNode->requiresExchangeClient() is true.

Reply:
Hi Masha, it cannot go pure virtual. That way every inherited translator will need to implement both toOperator methods, which for most operators, the one with exchangeClient will be returning nullptr anyways to indicate translation failure. That's why I let them by default be translation failure in the base class.

Reply:
Why can't we keep a single method, but add an exchangeClient parameter to it?

Reply:
Yes we can and have it default to nullptr (That was actually what I did previously) but that way many custom operators will have an unused /* exchangeClient */ parameter, involving external usages of custom translators as well. Overloading will not change current implementations by providing an additional API that can be used optionally.